ckanext/recombinant/logic.py

```diff
@@ -439,3 +439,20 @@ def recombinant_datastore_info(up_func, context, data_dict):
             }
 
     return info
+
+
+@chained_action
+def recombinant_datastore_upsert(up_func, context, data_dict):
+    """
+    Wraps datastore_upsert action to split Validation Errors with format_trigger_error.
+    """
+    try:
+        up_func(context, data_dict)
+    except ValidationError as e:
+        _error_dict = e.error_dict
```

Comment:
this doesn't do anything, if you're trying to *not* modify e.error_dict you can: 
```suggestion
        _error_dict = dict(e.error_dict)
        _error_dict['records'] = dict(_error_dict['records'])
```